data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/JacksonEvent.java

```diff
@@ -279,7 +279,15 @@ public String toJsonString() {
     @Override
     public String toJsonStringWithTags(final String tagsKey) {
         String result = jsonNode.toString();
-        return result.substring(0, result.length()-1) + ",\""+tagsKey+"\":"+ getMetadata().getTags()+"}";
+        String separator = "";
+
+        result = result.substring(0, result.length()-1) + ",\"tags\":[";
+        for (final String tag: getMetadata().getTags()) {
```

Comment:
This works as is but could be done with a stream like this

```
String tagSetString = String.join(",", tagSet
            .stream()
            .map(tagName -> ("\"" + tagName + "\""))
            .collect(Collectors.toList()));
```

which would mean only a single append (instead of one per tag) is needed. It will be more performant than appending a bunch of times

```
result += tagSetString
```